packages/sources/alphavantage-test/src/endpoint/forex.ts

```diff
@@ -0,0 +1,132 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { config } from '../config'
+import { PriceEndpoint, PriceEndpointParams } from '@chainlink/external-adapter-framework/adapter'
+
+const logger = makeLogger('Alphavantage HttpEndpoint')
+
+export const inputParameters = {
+  base: {
+    aliases: ['from', 'coin'],
+    type: 'string',
+    description: 'The symbol of symbols of the currency to query',
+    required: true,
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+    required: true,
+  },
+} as const
+
+export interface ProviderResponseBody {
+  'Realtime Currency Exchange Rate': {
+    '1. From_Currency Code': string
+    '2. From_Currency Name': string
+    '3. To_Currency Code': string
+    '4. To_Currency Name': string
+    '5. Exchange Rate': string
+    '6. Last Refreshed': string
+    '7. Time Zone': string
+    '8. Bid Price': string
+    '9. Ask Price': string
+  }
+  'Error Message': string
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: PriceEndpointParams
+  }
+  Response: SingleNumberResultResponse
+  Settings: typeof config.settings
+  Provider: {
+    RequestBody: never
+    ResponseBody: ProviderResponseBody
+  }
+}
+
+export const httpTransport = new HttpTransport<EndpointTypes>({
+  prepareRequests: (params, settings: typeof config.settings) => {
+    return params.map((param) => {
+      const from = param.base
+      const to = param.quote
+      const requestConfig = {
+        baseURL: settings.API_ENDPOINT,
+        method: 'GET',
+        params: {
+          function: 'CURRENCY_EXCHANGE_RATE',
+          from_currency: from,
+          to_currency: to,
+          from_symbol: from,
+          to_symbol: to,
+          symbol: from,
+          market: to,
+          apikey: settings.API_KEY,
+        },
+      }
+      return {
+        params,
```

Comment:
Also just realized we're passing all params here. I think we only want to pass the ones associated to this request. Otherwise in parseResponse we'll get all params for every response causing us to set the wrong results for the wrong params. Since each param has a separate request, I think this should be something like 
```suggestion
        params: [{ ...param }]
```

Reply:
I think we're still missing this change. Since we build requests for each individual param, we should only pass that param for this request
```suggestion
        params: [param],
```